src/test/java/org/junitpioneer/platform/testkit/engine/PioneerEngineExecutionResults.java

```diff
@@ -0,0 +1,156 @@
+				.engine("junit-jupiter")
+				.selectors(DiscoverySelectors.selectClass(testClass))
+				.execute();
+	}
+
+	public PioneerEngineExecutionResults(Class<?> testClass, String testMethodName) {
+		executionResults = EngineTestKit
+				.engine("junit-jupiter")
+				.selectors(DiscoverySelectors.selectMethod(testClass, testMethodName))
+				.execute();
+	}
+
+	/**
+	 * Get all recorded events.
+	 */
+	public Events all() {
+		return executionResults.all();
+	}
+
+	/**
+	 * Get recorded events for containers.
+	 *
+	 * <p>In this context, the word "container" applies to {@link TestDescriptor
+	 * TestDescriptors} that return {@code true} from {@link TestDescriptor#isContainer()}.</p>
+	 */
+	public Events containers() {
+		return executionResults.containers();
+	}
+
+	/**
+	 * Get recorded events for tests.
+	 *
+	 * <p>In this context, the word "test" applies to {@link TestDescriptor
+	 * TestDescriptors} that return {@code true} from {@link TestDescriptor#isTest()}.</p>
+	 */
+	public Events tests() {
+		return executionResults.tests();
+	}
+
+	/**
+	 * Returns the number of all tests.
+	 *
+	 * @return Number of all tests
+	 */
+	public long getTotalNumberOfTests() {
+		return executionResults.tests().count();
+	}
+
+	/**
+	 * Returns the number of failed tests.
+	 *
+	 * @return Number of failed tests
+	 */
+	public long getNumberOfFailedTests() {
+		return executionResults.all().failed().count();
+	}
+
+	/**
+	 * Returns the number of successful tests.
+	 *
+	 * @return Number of successful tests
+	 */
+	public long getNumberOfSucceededEvents() {
+		return executionResults.all().succeeded().count();
+	}
+
+	/**
+	 * Returns the number of skipped tests.
+	 *
+	 * @return Number of skipped tests
+	 */
+	public long getNumberOfSkippedEvents() {
+		return executionResults.all().skipped().count();
+	}
+
+	/**
+	 * Returns the number of aborted tests.
+	 *
+	 * @return Number of aborted events results
+	 */
+	public long getNumberOfAbortedEvents() {
+		return executionResults.all().aborted().count();
+	}
+
+	/**
+	 * Returns the message of the first failed event.
+	 * This can be used if you expect a test to fail with an exception and want to check the exception message.
+	 *
+	 * @return Message of the first failed event.
+	 */
+	public String getFirstFailuresThrowableMessage() {
+		return executionResults
+				.all()
+				.failed()
+				.stream()
+				.findFirst()
+				.orElseThrow(AssertionError::new)
+				.getPayload(TestExecutionResult.class)
+				.flatMap(TestExecutionResult::getThrowable)
+				.orElseThrow(AssertionError::new)
+				.getMessage();
+	}
+
+	/**
+	 * Returns the published report entries of all tests.
+	 *
+	 * @return published report entries of all tests
+	 */
+	public List<Map<String, String>> getPublishedTestReportEntries() {
+		return executionResults
+				.tests()
+				.reportingEntryPublished()
+				.stream()
+				.map(executionEvent -> executionEvent.getPayload(org.junit.platform.engine.reporting.ReportEntry.class))
```

Comment:
You could import this (in which case the import is used over the `ReportEntry` in the same package).